### ISSUE TITLE
Incompatibility with Pandas 0.19.x

### ISSUE THREAD

**Issue by jaksmid** [bmabey/pyLDAvis#76]
PyLDAvis worked within my environment with pandas 0.18.1.

After updating the pandas to 0.19.0. I get an error after loading my model:

```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-99591ceb694a> in <module>()
      1 import pyLDAvis
----> 2 vis_data = pyLDAvis.prepare(**data)
      3 pyLDAvis.display(vis_data)

/Users/smidj/.virtualenvs/lda/lib/python2.7/site-packages/pyLDAvis/_prepare.pyc in prepare(topic_term_dists, doc_topic_dists, doc_lengths, vocab, term_frequency, R, lambda_step, mds, n_jobs, plot_opts, sort_topics)
    396 
    397    topic_info         = _topic_info(topic_term_dists, topic_proportion, term_frequency, term_topic_freq, vocab, lambda_step, R, n_jobs)
--> 398    token_table        = _token_table(topic_info, term_topic_freq, vocab, term_frequency)
    399    topic_coordinates = _topic_coordinates(mds, topic_term_dists, topic_proportion)
    400    client_topic_order = [x + 1 for x in topic_order]

/Users/smidj/.virtualenvs/lda/lib/python2.7/site-packages/pyLDAvis/_prepare.pyc in _token_table(topic_info, term_topic_freq, vocab, term_frequency)
    265    # term-topic frequency table of unique terms across all topics and all values of lambda
    266    term_ix = topic_info.index.unique()
--> 267    term_ix.sort()
    268    top_topic_terms_freq = term_topic_freq[term_ix]
    269    # use the new ordering for the topics

/Users/smidj/.virtualenvs/lda/lib/python2.7/site-packages/pandas/indexes/base.pyc in sort(self, *args, **kwargs)
   1701 
   1702     def sort(self, *args, **kwargs):
-> 1703         raise TypeError("cannot sort an Index object in-place, use "
   1704                         "sort_values instead")
   1705 

TypeError: cannot sort an Index object in-place, use sort_values instead
```

If time permits, I will try to fix the code and prepare a pull request.
Regards,
Jakub

**Comment by jaksmid** [comment]
I see, this was already fixed in the recent pull request but was not released yet.
However, the fix uses numpy for sorting the values, I think the better option is to use the internal pandas method `term_ix.sort_values()` instead of `term_ix = np.sort(term_ix.values)`. 
I am happy to update the code if this change is desired. Otherwise, feel free to close the issue.

Thanks,
Jakub

**Comment by bmabey** [comment]
I wondered about that myself but I didn't know how much that would matter.  I'd take a PR if you're up to it. Thanks!

**Comment by bmabey** [comment]
FYI, it looks like using `sort_values()` would be preferred. https://github.com/bmabey/pyLDAvis/issues/74#issuecomment-254250278

**Comment by jaksmid** [comment]
Agreed, I will create a pull requests in few days.
Want to double check that we get the same results with different versions of pandas.

**Comment by parthoiiitm** [comment]
Hi, the method ```sort_values()``` is working with pandas version 0.19 but isn't working with 0.18.0 ,  had to update pandas

**Comment by jaksmid** [comment]
Thanks @parthoiiitm  for the report.
I have verified that the sort values does not work with pandas 0.18.0.
In this case I think it is better to use the current fix for maintaining the backwards compatibility.
Close this issue then?

**Comment by bmabey** [comment]
From @parthoiiitm's report the current fix still has issues with `0.18.0` so the correct fix perhaps would drop the values so: `term_ix = np.sort(term_ix)`. I'll try this out before releasing it.

**Comment by amirfz** [comment]
has this been resolved yet? it seems to give me the same error even with the hacks suggested here

`/usr/local/lib/python3.5/dist-packages/pyLDAvis/_prepare.py in _token_table(topic_info, term_topic_freq, vocab, term_frequency)
    265    # term-topic frequency table of unique terms across all topics and all values of lambda
    266    term_ix = topic_info.index.unique()
--> 267    term_ix = np.sort(term_ix)
    268    top_topic_terms_freq = term_topic_freq[term_ix]
    269    # use the new ordering for the topics

/usr/local/lib/python3.5/dist-packages/pandas/indexes/base.py in sort(self, *args, **kwargs)
   1731 
   1732     def sort(self, *args, **kwargs):
-> 1733         raise TypeError("cannot sort an Index object in-place, use "
   1734                         "sort_values instead")
   1735 

TypeError: cannot sort an Index object in-place, use sort_values instead`

I'm using pandas 0.19 and pyLDAvis 2.0.0

**Comment by bmabey** [comment]
I've pushed a version that worked for me in pandas 0.18 and 0.19. It has also been released as `2.1.0`. Please try it out and report back if the problem persists.
